### ISSUE TITLE
Can u tell me how to address the issue of spatialsearch

### ISSUE THREAD

**Issue by QUTGXX** [yiyuan1991/3D-Face-GCNs#3]
It shows that ImportError: cannot import name 'spatialsearch' from 'lib'

**Comment by HOMGH** [comment]
> It shows that ImportError: cannot import name 'spatialsearch' from 'lib'

I have the same problem. There is just a "spatialsearch...... .so" file in lib folder!

**Comment by FuxiCV** [comment]
Sorry for the inconvenience, the "spatialsearch" comes from the "Perceiving Systems Mesh Package" (https://github.com/MPI-IS/mesh) mentioned in the readme file.
The "spatialsearch...... .so" was compiled under python 3.5 and Linux, if you want to run on other environments, you may refer to https://github.com/MPI-IS/mesh and install the "Perceiving Systems Mesh Package".

**Comment by QUTGXX** [comment]
> Sorry for the inconvenience, the "spatialsearch" comes from the "Perceiving Systems Mesh Package" (https://github.com/MPI-IS/mesh) mentioned in the readme file.
> The "spatialsearch...... .so" was compiled under python 3.5 and Linux, if you want to run on other environments, you may refer to https://github.com/MPI-IS/mesh and install the "Perceiving Systems Mesh Package".

Thanks for your reply, I have tried to install the package followed by the steps mentioned on the Github.  When I tried to 'make tests', its successes.  But 'spatialsearch' still not work.

**Comment by HOMGH** [comment]
> Sorry for the inconvenience, the "spatialsearch" comes from the "Perceiving Systems Mesh Package" (https://github.com/MPI-IS/mesh) mentioned in the readme file.
> The "spatialsearch...... .so" was compiled under python 3.5 and Linux, if you want to run on other environments, you may refer to https://github.com/MPI-IS/mesh and install the "Perceiving Systems Mesh Package".

Thank you for your quick reply. I'm actually working on Python3.5 and Linux. So I don't need to install "Perceiving Systems Mesh Package" again? But how can I import spatialsearch? because its not .py, its .so. 
Should I convert this "spatialsearch.... .so" and also "aabb_normals.... .so" to .py file?

**Comment by FuxiCV** [comment]
Hey, if you install the "Perceiving Systems Mesh Package" successfully, then "import spatialsearch" is not needed. Most of the mesh sampling code is borrowed from CoMA (https://github.com/anuragranj/coma/blob/master/lib/mesh_sampling.py), we copy the two .so files to lib folder just for convenience.
You can directly import those .so files if the OS and python version are matched, no need to convert to .py file.

**Comment by QUTGXX** [comment]
> Hey, if you install the "Perceiving Systems Mesh Package" successfully, then "import spatialsearch" is not needed. Most of the mesh sampling code is borrowed from CoMA (https://github.com/anuragranj/coma/blob/master/lib/mesh_sampling.py), we copy the two .so files to lib folder just for convenience.
> You can directly import those .so files if the OS and python version are matched, no need to convert to .py file.

Got it, thanks for your reply. Can u plz tell me the link of input data such as bfm09_face_template.obj and CelebA?

**Comment by HOMGH** [comment]
> > Hey, if you install the "Perceiving Systems Mesh Package" successfully, then "import spatialsearch" is not needed. Most of the mesh sampling code is borrowed from CoMA (https://github.com/anuragranj/coma/blob/master/lib/mesh_sampling.py), we copy the two .so files to lib folder just for convenience.
> > You can directly import those .so files if the OS and python version are matched, no need to convert to .py file.
> 
> Got it, thanks for your reply. Can u plz tell me the link of input data such as bfm09_face_template.obj and CelebA?

Hi. Sorry I still have problems in importing .so files. The OS and Python version are matched. So how can I import those .so files?
Thanks.

**Comment by deepmo24** [comment]
@QUTGXX @HOMGH Hi，have you solved this problem? I installed the "Perceiving Systems Mesh Package" , too.  However, how to solve the "spatialsearch" import error? I would be very appreciated if you could reply!

**Comment by deepmo24** [comment]
@FuxiCV Hi, thanks for your great project. In your reply, you said there is no need to "import spatialsearch" when install the  "Perceiving Systems Mesh Package". However, in `utils.py`, we still needs to call "spatialsearch.aabbtree_compute()". https://github.com/FuxiCV/3D-Face-GCNs/blob/a12d8ba6a328217d3dcc681cbb256bd8b860b290/utils.py#L650
So how to solve  this problem? Thank you!

**Comment by deepmo24** [comment]
I solved it by `from psbody.mesh import spatialsearch` after installing the "Perceiving Systems Mesh Package".

**Comment by HOMGH** [comment]
> I solved it by `from psbody.mesh import spatialsearch` after installing the "Perceiving Systems Mesh Package".

Cool! Could you train the model as well?

**Comment by deepmo24** [comment]
@HOMGH No, still facing some problems. please refer to https://github.com/FuxiCV/3D-Face-GCNs/issues/10#issuecomment-672974312

**Comment by HOMGH** [comment]
> @HOMGH No, still facing some problems. please refer to [#10 (comment)](https://github.com/FuxiCV/3D-Face-GCNs/issues/10#issuecomment-672974312)

I think this (#7) would help.

**Comment by deepmo24** [comment]
> > @HOMGH No, still facing some problems. please refer to [#10 (comment)](https://github.com/FuxiCV/3D-Face-GCNs/issues/10#issuecomment-672974312)
> 
> I think this (#7) would help.

Hi, not these files.

**Comment by lj6052317** [comment]
> I solved it by `from psbody.mesh import spatialsearch` after installing the "Perceiving Systems Mesh Package".
While installing the "Perceiving Systems Mesh Package",I cannot build successful.

**Comment by cyjouc** [comment]
> > Hey, if you install the "Perceiving Systems Mesh Package" successfully, then "import spatialsearch" is not needed. Most of the mesh sampling code is borrowed from CoMA (https://github.com/anuragranj/coma/blob/master/lib/mesh_sampling.py), we copy the two .so files to lib folder just for convenience.
> > You can directly import those .so files if the OS and python version are matched, no need to convert to .py file.
> 
> Got it, thanks for your reply. Can u plz tell me the link of input data such as bfm09_face_template.obj and CelebA?

Hi,could you share the input data  processing code with me?Txs o lot

**Comment by cyjouc** [comment]
> CelebA

Hi,Coule you solve the problem for CelebA preprocessing dataset?

**Comment by AaryanAshok** [comment]
> I solved it by `from psbody.mesh import spatialsearch` after installing the "Perceiving Systems Mesh Package".

Please explain the procedure, I have installed the "Perceiving Systems Mesh Package" still unable to solve the import error is it the ".so" file or ".py" file and how to import after installation of "Perceiving Systems Mesh Package".
